core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -269,28 +269,28 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
   }
 
   private def initZk(): ZkUtils = {
-    info("Connecting to zookeeper on " + config.zkConnect)
+    info(s"Connecting to zookeeper on ${config.zkConnect}")
 
-    val chroot = {
-      if (config.zkConnect.indexOf("/") > 0)
-        config.zkConnect.substring(config.zkConnect.indexOf("/"))
-      else
-        ""
+    val chrootIndex = config.zkConnect.indexOf("/")
+    val chrootOption = {
+      if (chrootIndex > 0) Some(config.zkConnect.substring(chrootIndex))
+      else None
     }
 
-    val secureAclsEnabled = JaasUtils.isZkSecurityEnabled() && config.zkEnableSecureAcls
+    val secureAclsEnabled = config.zkEnableSecureAcls
```

Comment:
I think I'm confused about the original bug report. If `zookeeper.set.acl` is `true` but ZK security isn't enabled, then `secureAclsEnabled` will be `false && true` -> `false` and the if statement condition will be `true && !false` -> `true` and the exception will be thrown. Am I confused about something in the existing code? I don't see why the exception wouldn't be thrown.